disco_aws_automation/disco_aws.py

```diff
@@ -355,7 +356,8 @@ def provision(self, ami, hostclass=None,
 
         group = self.autoscale.get_group(
             hostclass=hostclass, launch_config=launch_config.name,
-            vpc_zone_id=",".join([subnet.id for subnet in self.get_subnets(meta_network, hostclass)]),
+            vpc_zone_id=",".join([subnet['SubnetId'] for subnet
+                                  in self.get_subnets(meta_network, hostclass)]),
             min_size=size_as_minimum_int_or_none(min_size),
             max_size=size_as_maximum_int_or_none(max_size),
             desired_size=size_as_maximum_int_or_none(desired_size),
```

Comment:
A more general suggestion for this file: it looks like disco_aws.py is 800 lines, which is vaguely shocking to me (I've seen Java files that big, but those are also shocking, and Python likes to talk about how compact it is).  Is it possible we could, either as part of this project or as a short-term refactor, get all or some of the network utilties into a separate, more-coherent package?